### PR TITLE
search: remove onboarding tour

### PR DESCRIPTION
**Pull request by novoselrok**
Removes all the code powering the search onboarding tour and the Shepherd.js library.

## Test plan

* Create a brand new user and try searching
* No tour should appear

## App preview:

- [Web](https://sg-web-rn-remove-search-onboarding-tour.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-gavojalymp.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by fkling** [comment]
After my hard work making it work with CodeMirror?? Just kidding, less code is better code 🚀

**Review by fkling — APPROVED** [review]

File: client/web/src/user/settings/UserSettingsSidebar.tsx [hunk 1]
```diff
@@ -45,8 +44,6 @@ export interface UserSettingsSidebarProps
 export const UserSettingsSidebar: React.FunctionComponent<
     React.PropsWithChildren<UserSettingsSidebarProps>
 > = props => {
-    const [, setHasCancelledTour] = useTemporarySetting('search.onboarding.tourCancelled')
-    const showOnboardingTour = useExperimentalFeatures(features => features.showOnboardingTour ?? false)
```
Comment (limitedmage) [inline client/web/src/user/settings/UserSettingsSidebar.tsx:49]: Should we mark this feature as REMOVED in the [settings schema](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/schema/settings.schema.json?L120%3A1-128%3A1=) and remove all uses of it? It's prop-drilled through quite a few places.

Comment (novoselrok) [reply client/web/src/user/settings/UserSettingsSidebar.tsx:49]: You're right, I'll deprecate it and remove the leftover usage.

File: client/web/src/search/home/SearchPageInput.tsx [hunk 2]
```diff
@@ -145,12 +135,9 @@ export const SearchPageInput: React.FunctionComponent<React.PropsWithChildren<Pr
         <div className="d-flex flex-row flex-shrink-past-contents">
             <Form className="flex-grow-1 flex-shrink-past-contents" onSubmit={onSubmit}>
                 <div data-search-page-input-container={true} className={styles.inputContainer}>
-                    {/* Search onboarding tour must be rendered before the SearchBox so
-                    the Monaco autocomplete suggestions are not blocked by the tour. */}
                     <div ref={tourContainer} />
```
Comment (limitedmage) [inline client/web/src/search/home/SearchPageInput.tsx:138]: Can we remove this div?

File: schema/settings.schema.json [hunk 3]
```diff
@@ -118,7 +118,7 @@
           }
         },
         "showOnboardingTour": {
-          "description": "Enables the onboarding tour.",
+          "description": "REMOVED. Previously, it enabled the onboarding tour.",
```
Comment (limitedmage) [inline schema/settings.schema.json:121]: I think at this point, if a feature is removed (as opposed to permanently enabled), it should have anything else in the description. See apiDocs for example. Keeps users from thinking "what if".

**Review by limitedmage — APPROVED** [review]
Looks great! Just a couple more suggestions.

**Pull request merged**